File: examples/javascript/variables/variables.js
// 第 08 章 · 变量 — JavaScript 示例
// 运行：node variables.js

// 1. 声明
let studentName = "Alice";
const MAX_SCORE = 100;
let age = 20;
let score = 92;
console.log(studentName, age, score, MAX_SCORE);

// 2. 类型跟着值走
console.log("typeof score:", typeof score);
score = "A+";
console.log("typeof score:", typeof score);

// 3. 赋值语义：原始类型复制值
let a = 92;
let b = a;
b = 60;
console.log("原始类型 值复制:", a, b);        // 92 60

// 4. 赋值语义：对象复制引用
let s1 = { name: "Alice", score: 92 };
let s2 = s1;
s2.score = 60;
console.log("对象 引用复制:", s1.score, s2.score);  // 60 60

// 5. const 锁定绑定而非内容
const student = { score: 92 };
student.score = 60;            // 合法
console.log("const 对象可改内容:", student.score);

File: examples/python/variables/variables.py
# 第 08 章 · 变量 — Python 示例
# 运行：python3 variables.py

# 1. 赋值即绑定
student_name = "Alice"
MAX_SCORE = 100        # 约定的"常量"
age = 20
score = 92
print(student_name, age, score, MAX_SCORE)

# 2. 类型跟着值走
print("type:", type(score).__name__)
score = "A+"
print("type:", type(score).__name__)

# 3. 不可变对象：重新绑定
a = 92
b = a
b = 60
print("不可变对象 重新绑定:", a, b)          # 92 60

# 4. 可变对象：共享同一对象
s1 = {"name": "Alice", "score": 92}
s2 = s1
s2["score"] = 60
print("可变对象 共享绑定:", s1["score"])      # 60
print("是同一对象吗:", s1 is s2)             # True

# 5. 正确的复制方式
s3 = s1.copy()
s3["score"] = 100
print("复制后互不影响:", s1["score"], s3["score"])  # 60 100
